account_bank_statement_import_online_ponto/models/online_bank_statement_provider_ponto.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2020 Florent de Labarre
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import requests
+import json
+import base64
+import time
+import pytz
+import re
+from datetime import datetime
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from dateutil.relativedelta import relativedelta
+from odoo.addons.base.models.res_bank import sanitize_account_number
+
+PONTO_ENDPOINT = 'https://api.myponto.com'
+
+
+class OnlineBankStatementProviderPonto(models.Model):
+    _inherit = 'online.bank.statement.provider'
+
+    ponto_token = fields.Char(readonly=True)
+    ponto_token_expiration = fields.Datetime(readonly=True)
+    ponto_last_identifier = fields.Char(readonly=True)
+
+    def ponto_reset_last_identifier(self):
+        self.write({'ponto_last_identifier': False})
+
+    @api.model
+    def _get_available_services(self):
+        return super()._get_available_services() + [
+            ('ponto', 'MyPonto.com'),
+        ]
+
+    def _obtain_statement_data(self, date_since, date_until):
+        self.ensure_one()
+        if self.service != 'ponto':
+            return super()._obtain_statement_data(
+                date_since,
+                date_until,
+            )
+        return self._ponto_obtain_statement_data(date_since, date_until)
+
+    def _get_statement_date(self, date_since, date_until):
+        self.ensure_one()
+        if self.service != 'ponto':
+            return super()._get_statement_date(
+                date_since,
+                date_until,
+            )
+        return date_since.astimezone(pytz.timezone('Europe/Paris')).date()
```

Comment:
Why do you convert UTC to other timezone? Also, it's better to return datetime to base module for more correct sorting

Reply:
If you have an transaction with this datetime : `2020-04-16 - 23:30:00 UTC`, In France UTC+2, the datetime is `2020-04-17 - 1:30:00 UTC+2`, the accounting date is the **17** of april, and not the 16 of april. (Statement work with date and not date_time)

I think there is an issue on your code when you replace hours, minute by zero to find the date, it is not the correct way to find the date. On each provider the tz should be set (when provider send date in UTC format), to convert date_time to date.

Ponto/Qonto are based in Europe/Paris, that why I convert in tz `Europe/Paris` to find the good date.

Reply:
Correct me if I'm wrong, but you should be using timezone of company, not timezone of provider, since if the provider would be PayPal, doubtfully you'd use the US EST timezone, right?

Reply:
the company have not a `tz` field.

It is more logical to use the timezone of the bank, like that you have the same date between odoo and the bank paper statement (date of bank paper statement == date of accounting).

I think for Paypal it is different, because it is not a bank. (keep the user chosse)

Reply:
When you get statement from Ponto in form of PDF, what timezone it uses?

Reply:
> date of bank paper statement == date of accounting

I'm not confident about this, yet I'm no expert to claim what's right or wrong in this case. Odoo statement import module always uses the date as-is supplied by the provider, w/o conversions, if I recall correctly. @pedrobaeza maybe you have some deeper knowledge on the matter?

Reply:
Ponto is not a bank, there are no PDF, it agregate data from banks.

On PDF Qonto only the date is written (but by the api you have date and time), Qonto is based in France.

Reply:
and the PDF in Qonto contains date that corresponds the API response around the midnight or requires conversion to French timezone?

Reply:
For this transaction : "2020-04-15T18:22:30.296Z", I was created at 20h22 (French Timezone).

for exemple in OFX file the timezone is write. https://www.ibm.com/developerworks/xml/library/x-ofxv2/

With cfonb file (french horrible bank file) there is only the date.

Reply:
I don't understand what do you want from me

Reply:
:D A slice of common sense, as usual! The debate is over this: if provider provides transaction timestamp in UTC full date-time format, what date should be saved in statement? Options:

- UTC date
- Date from a timestamp in assumed timezone of provider (Europe/Paris in this case)
- Your option :)

Reply:
Frankly speaking, I can find pros and cons for each option so far, because usually API returns in UTC time, but PDF from same service can sometimes be in some fixed timezone or users timezone, or etc. timezone. Maybe it would be even worth adding an option "convert to XXX timezone"

Reply:
It is an accounting question:
Some banks doesn't give a `date` of transaction but a `datetime`, and some time without the timezone (just in UTC).
In Odoo the statement use only the `date`.

Question : how convert the `datetime` to `date` (use the TZ of company, TZ of bank, TZ of user) ?

I think we should use the TZ of bank, because on the PDF bank report there is the date at the bank TZ.

Reply:
Ultimately, we can give user option to select, since otherwise it's just an assumption, any of options.

Reply:
I've created #281 to expose timezone configuration to the user in order to avoid such magic within the provider. If the mentioned PR would me merged, provider should return with whatever timezone it has (naive datetime will be treated as UTC)

Reply:
I'm getting this error trying to fetch things from Ponto:
```
ValueError: astimezone() cannot be applied to a naive datetime
```
Please remove this conversion